test/test_dns.py

```diff
@@ -63,18 +71,30 @@ def run_test(self):
                         'readPreferenceTags', opts.pop('readpreferencetags'))
                     opts['readPreferenceTags'] = rpts
                 self.assertEqual(result['options'], options)
+            if parsed_options:
+                for opt in parsed_options:
+                    if opt == 'user':
+                        self.assertEqual(result['username'],
+                                         parsed_options[opt])
+                    elif opt == 'password':
+                        self.assertEqual(result['password'],
+                                         parsed_options[opt])
+                    elif opt == 'auth_database' or opt == 'db':
+                        self.assertEqual(result['database'],
+                                         parsed_options[opt])
 
             hostname = next(iter(client_context.client.nodes))[0]
             # The replica set members must be configured as 'localhost'.
             if hostname == 'localhost':
                 copts = client_context.default_client_options.copy()
+                # Remove tls since SRV parsing should add it automatically.
+                copts.pop('tls', None)
```

Comment:
Doesn't this pose another problem? Since kwargs override opts passed in the URI, we might be overriding what the test wants to pass? Should we check the keys in `result` to ensure that none of them matches keys in `copts`?